TELCODOCS-28: New status items, drainProgress and elapsedTime, added (#46914)
openshift/openshift-docs: nodes/nodes/eco-node-maintenance-operator.adoc

toc::[]

> You can use the Node Maintenance Operator to place nodes in maintenance mode.

Reviewer: > > I would like to delete the second sentence in the beginning of the page - http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html `This is a standalone version of the Node Maintenance Operator that is independent of OpenShift Virtualization installation.`
> 
> @razo7 This is now updated, see http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

Now the `You can use the Node Maintenance Operator to place nodes in maintenance mode.` sentence doesn't add anything, that we don't say under `About the Node Maintenance Operator`.
I think it would be better to delete the whole sentence and the about section, `About the Node Maintenance Operator`,  and let the page starts with `You can place nodes into maintenance mode using the oc adm utility, or using NodeMaintenance custom resources (CRs).`

Author: Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html

[NOTE]
====

> If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version. You must first upgrade {VirtProductName}, and then install the latest version of the Node Maintenance Operator from the `OperatorHub`.

Reviewer: I would move the NOTE from the beginning to be under [Installing the Node Maintenance Operator](http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-standalone-nmo) since it is about installing.

Reviewer: By looking in https://issues.redhat.com/browse/CNV-17623?focusedCommentId=20673382&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20673382 I see that CNV suggest that a user can and should use "our" NMO before doing an upgrade if he/she want the node to stay under maintenance while doing the CNV upgrade...

I would suggest to leave the note, without an action from the user, as follows:

> If OpenShift Virtualization v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.

Author: Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#installing-standalone-nmo

toc::[]

> You can use the Node Maintenance Operator to place nodes in maintenance mode, by using the `oc adm` utility, or `NodeMaintenance` custom resources (CRs).

Reviewer 2: ```suggestion
You can use the Node Maintenance Operator to place nodes in maintenance mode by using the `oc adm` utility or `NodeMaintenance` custom resources (CRs).
```

Author: Thanks @jeana-redhat , updated as suggested

include::modules/eco-about-node-maintenance-standalone.adoc[leveloffset=+1]

[id="installing-standalone-nmo"]
== Installing the Node Maintenance Operator
You can install the Node Maintenance Operator using the web console or the OpenShift CLI (`oc`).

[NOTE]
====

> If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.

Reviewer 2: Do they need to do anything about it in this case?

```suggestion
If {VirtProductName} version 4.10 or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator.
```

Author: Thanks @jeana-redhat , updated as suggested

Author: No, no action required. 

Feedback from SME earlier:
By looking in https://issues.redhat.com/browse/CNV-17623?focusedCommentId=20673382&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20673382 I see that CNV suggest that a user can and should use "our" NMO before doing an upgrade if he/she want the node to stay under maintenance while doing the CNV upgrade...

I would suggest to leave the note, without an action from the user, as follows:

If OpenShift Virtualization v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version.

[NOTE]
====

> If {VirtProductName} v4.10.z or less is installed in your cluster, it includes an outdated version of the Node Maintenance Operator version. Upgrade {VirtProductName} to avail of the latest version of the Node Maintenance Operator.

Reviewer 3: s/avail of/obtain/

Author: Thanks John, I've included this feedback.

[id="setting-node-in-maintenance-mode"]
== Setting a node to maintenance mode
You can place a node into maintenance mode from the web console or from the CLI by using a `NodeMaintenance` CR.

include::modules/eco-setting-node-maintenance-cr-web-console.adoc[leveloffset=+2]

include::modules/eco-setting-node-maintenance-cr-cli.adoc[leveloffset=+2]

include::modules/eco-checking_status_of_node_maintenance_cr_tasks.adoc[leveloffset=+2]

[id="resuming-node-from-maintenance-mode"]
== Resuming a node from maintenance mode
You can resume a node from maintenance mode from the web console or from the CLI by using a `NodeMaintenance` CR. Resuming a node brings it out of maintenance mode and makes it schedulable again.

include::modules/eco-resuming-node-maintenance-cr-web-console.adoc[leveloffset=+2]

include::modules/eco-resuming-node-maintenance-cr-cli.adoc[leveloffset=+2]

[id="working-with-bare-metal-nodes"]
== Working with bare-metal nodes

> For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode, by using the web console Actions control.

Reviewer 2: ```suggestion
For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode, by using the web console *Actions* control.
```

Author: Thanks @jeana-redhat , updated as suggested

[NOTE]
====

> Clusters with bare-metal nodes can also place a node into maintenance mode, and resume a node from maintenance mode, by using the web console and CLI, as outlined. These methods, by using the web console Actions control, are applicable to bare-metal clusters only.

Reviewer 2: ```suggestion
Clusters with bare-metal nodes can also place a node into maintenance mode, and resume a node from maintenance mode, by using the web console and CLI, as outlined. These methods, by using the web console *Actions* control, are applicable to bare-metal clusters only.
```

Author: Thanks @jeana-redhat , updated as suggested

== Gathering data about the Node Maintenance Operator
To collect debugging information about the Node Maintenance Operator, use the `must-gather` tool. For information about the `must-gather` image for the Node Maintenance Operator, see xref:../../support/gathering-cluster-data.adoc#gathering-data-specific-features_gathering-cluster-data[Gathering data about specific features].

[id="working-with-bare-metal-nodes"]
== Working with bare-metal nodes

> In addition to maintaining virtual nodes, the Node Maintenance Operator can also maintain bare-metal nodes. For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode by using the web console Actions control.

Reviewer: For clusters with bare-metal nodes, you can place a node into maintenance mode, and resume a node from maintenance mode by using the web console Actions control. These methods are additional to the above methods and only valid for bare-metal cluster.

Reviewer: Then add a NOTE

> These methods, setting and resuming `NodeMaintenance` CR, are additional to the above methods and only valid for bare-metal cluster.

Reviewer: @ogradyp @anna-savina WDYT?
I think that without this note the user might think that only the methods under `Working with bare-metal nodes` are valid for bare-metal nodes/cluster.

Author: Updated as requested http://file.emea.redhat.com/pogrady/TELCODOCS-28/nodes/nodes/eco-node-maintenance-operator.html#working-with-bare-metal-nodes
